### PR TITLE
Removes job_access_type, adds directly access set after job equip

### PR DESCRIPTION
**Pull request by quardbreak**

### CODE REVIEW

File: maps/ministation/outfits/command.dm [hunk 1]
```diff
@@ -27,17 +27,13 @@
 			else
 				qdel(medal)
 
-/obj/item/card/id/gold/ministation
-	job_access_type = /datum/job/ministation/captain
+/obj/item/card/id/gold
```
Comment (MistakeNot4892) [inline maps/ministation/outfits/command.dm]: Why leave this stub?

Comment (quardbreak) [reply maps/ministation/outfits/command.dm]: Oopsie

File: code/game/objects/items/weapons/cards_ids.dm [hunk 2]
```diff
@@ -181,18 +179,6 @@ var/const/NO_EMAG_ACT = -50
...
-	.=..()
-	if(job_access_type)
-		var/datum/job/j = SSjobs.get_by_path(job_access_type)
-		if(j)
-			rank = j.title
-			assignment = rank
-			access |= j.get_access()
-			if(!detail_color)
-				detail_color = j.selection_color
-	update_icon()
```
Comment (MistakeNot4892) [inline code/game/objects/items/weapons/cards_ids.dm:194]: Should still call update_icon as it's indepedant of access setting and needed by stuff like the captain's spare card.

File: code/game/machinery/computer/guestpass.dm [hunk 3]
```diff
@@ -91,7 +91,7 @@
 
 	if(giver)
 		data["giver"] = !!giver
-		data["giver_name"] = giver.rank || giver.assignment || SSjobs.get_by_path(giver.job_access_type).title
+		data["giver_name"] = giver.rank || giver.assignment || SSjobs.get_by_title(giver.rank)
```
Comment (MistakeNot4892) [inline code/game/machinery/computer/guestpass.dm]: This line is redundant, `giver.rank` will short-circuit before it ever hits the SSjobs call. You can just delete `SSjobs.get_by_title(giver.rank)` entirely.

**Review by MistakeNot4892 — APPROVED** [review]

**Pull request merged**